temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -245,9 +249,43 @@ private StartWorkflowExecutionRequest overrideStartWorkflowExecutionRequest(
     if (taskTimeoutMillis != Durations.toMillis(request.getWorkflowTaskTimeout())) {
       request.setWorkflowTaskTimeout(Durations.fromMillis(taskTimeoutMillis));
     }
+
+    request.setSearchAttributes(getValidSearchAttributes(request));
     return request.build();
   }
 
+  private SearchAttributes.Builder getValidSearchAttributes(
+      StartWorkflowExecutionRequest.Builder request) {
+
+    SearchAttributes.Builder searchAttributes = request.getSearchAttributes().toBuilder();
+    Map<String, SearchAttributesUtil.Pair<String, SearchAttributeType>> registeredAttributes =
+        store.getRegisteredSearchAttribute();
+
+    for (Map.Entry<String, Payload> searchAttribute :
+        searchAttributes.getIndexedFieldsMap().entrySet()) {
+
+      String key = searchAttribute.getKey();
+      Payload val = searchAttribute.getValue();
+
+      if (!registeredAttributes.containsKey(key)) {
+        throw Status.INVALID_ARGUMENT
+            .withDescription(key + " is not a valid search attribute name")
+            .asRuntimeException();
+      } else {
+        SearchAttributeType type = registeredAttributes.get(key).getType();
```

Comment:
This codeblock and logic don't make too much sense to me, I'm not sure that it even has a consumer. Let's discuss on Zoom.